src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -678,10 +678,10 @@ public bool IsObjectInitializerNamedAssignmentIdentifier([NotNullWhen(true)] Syn
                 identifier.IsLeftSideOfAssignExpression() &&
                 identifier.Parent.IsParentKind(SyntaxKind.ObjectInitializerExpression))
             {
-                var objectInitializer = identifier.Parent.Parent;
-                if (objectInitializer.IsParentKind(SyntaxKind.ObjectCreationExpression))
+                var objectInitializer = identifier.Parent.GetRequiredParent();
```

Comment:
Why is just the last one GetRequiredParent, should both `.Parent` be GetRequiredParent?  And same below

Reply:
the first is known to be not-null because `identifier.Parent.IsParentKind` returned true.  it is NRT annotated, so the compiler knows that it is not null.